### PR TITLE
Add explicit snapshot date subset to `Widget_GroupOverview`.

### LINKED ISSUE #1787: Feature: Subset `dfResults` argument to `Widget_GroupOverview` on latest snapshot date.

### LINKED ISSUE #1788: Feature: Add parameter to `Report_MetricTable` that controls how `dfResults` is subset.
## Feature Details
i.e. amber + red flags (current behavior), any flag, or only red flags.

### PR DESCRIPTION
**Pull request by samussiah**
## Overview
- Resolves #1787 
- Resolves #1788 

## Test Notes/Sample Code
```
load_all()
Widget_GroupOverview(
  dfResults = filter_by_latest_SnapshotDate(reportingResults),
  dfMetrics = reportingMetrics,
  dfGroups = reportingGroups
)

Report_MetricTable(
  dfResults = reportingResults %>%
      dplyr::filter(.data$MetricID == 'kri0001') %>%
      filter_by_latest_SnapshotDate(),
  dfGroups = reportingGroups
)
```

Notes: 
@jonthegeek - #1788 ties into [this issue in {gsm.app}](https://github.com/Gilead-BioStats/gsm.app/issues/176).  Allows you to generate the analysis output table with all groups in addition to only groups flagged for the metric.

### CODE REVIEW

**Comment by jonthegeek** [comment]
@samussiah I *think* what we really want for that ticket is to show all of the results, but *highlight* the selected site (similar to what we do with the plots). Is that on your radar?

File: R/Widget_GroupOverview.R [hunk 1]
```diff
@@ -53,12 +53,24 @@ Widget_GroupOverview <- function(
...
+    cli::cli_alert_warning(
+      "[ dfResults ] contains multiple snapshot dates and will be subset on the latest snapshot date: {latest_snapshot_date}."
+    )
+
+    dfResults <- dfResults %>%
+      dplyr::filter(
+        .data$SnapshotDate == latest_snapshot_date
+      )
+  }
+
```
Comment (jonthegeek) [inline R/Widget_GroupOverview.R:73]: I made a convenience wrapper, `filter_by_latest_SnapshotDate()`, to handle & standardize this. Adding the warning when we apply that would probably be generally useful to add in `filter_by_latest_SnapshotDate()`, though (but only when `length(unique(dfResults$SnapshotDate > 1)`).

**Review by jonthegeek — CHANGES_REQUESTED** [review]
It'd be good to standardize on `filter_by_latest_SnapshotDate()` so we can use the same messaging everywhere that it's used.

File: R/Widget_GroupOverview.R (hunk 1 above)
Comment (samussiah) [reply R/Widget_GroupOverview.R]: Anywhere we're filtering on latest snapshot date has been updated to use this function.

**Review by jonthegeek — APPROVED** [review]
Looks good to me!

**Review by lauramaxwell — APPROVED** [review]
working well for me, both in standalone function call and in reporting workflow.

**Pull request merged**